lib/yggdrasil/commands.ex

```diff
@@ -0,0 +1,110 @@
+defprotocol Yggdrasil.Command do
+  def execute(player, cmd)
+end
+
+defmodule Yggdrasil.Command.MoveCommand do
+  defstruct exit: nil
+end
+
+defimpl Yggdrasil.Command, for: Yggdrasil.Command.MoveCommand do
```

Comment:
Yeah at the time I wasn't sure and didn't want to try. Was too concerned about figuring out how I wanted things to look and work. I'll try to split it out when I have something working.